W-13498832-date-and-time-update (#184)
mulesoft/docs-rpa: rpa-builder/6.3/modules/ROOT/pages/toolbox-variable-handling-generate-and-modify-date-and-time.adoc

other _Action Steps_.

== Properties

* *Generate Date and Time Settings*
+
_Generate date and time_ incorporates the specified values into the calculation of the selected time
stamp (time offset). This action step supports positive and negative numbers. The
default value is 0 in each case.
* *Calendar week setting* 
+
Specify either Monday or Sunday as the starting day for the week.
* *Days to Add* 
+
Number of days to add.
* *Generate day and time mode*
+
Select whether to use the *Current system date and time*, *Custom date and time*, or *Generated date and time from given variables*. The time offsets from the *Years to add*, *Months to add*, *Weeks to add*, *Days to add*, *Hours to add*, *Minutes to add*, *Seconds to add* fields apply for the calculation in all cases.
* *Hours to add* 
+
Number of hours to add.
* *Minutes to add* 
+
Number of minutes to add.
* *Months to add* 
+
Number of months to add.
* *Output language culture (for the day of the week)* 
+
Specify the output language to use for the days of the week. 
* *Override activated*
+
Enables overriding workday settings. 
* *Override to day* 
+

> Specify the day to override with. 

Reviewer: ```suggestion
Specify the day to override. 

```

* *Override to first or last* 
+

> Specify to use either the first or last day of the given period (for example, last day of the month).

Reviewer: ```suggestion
Specify either the first or last day of the given period (for example, last day of the month).

```

* *Seconds to add* 
+
Number of seconds to add.
* *Start workday of week*
+

> Specify the starting working day of the week.

Reviewer: ```suggestion
Specify the starting workday of the week.

```

* *Weeks to add* 
+
Number of weeks to add.
* *Workdays setting* 
+

> Select the length of the workweek between *5 days workweek* and *6 days workweek*.

Reviewer: Is this in the UI like this? It would normally be five-day workweek, six-day workweek.

* *Years to add* 
+
Number of years to add.

== Variables

> The following variables relate to the configurable time stamp generated after applying any

Reviewer: ```suggestion
The following variables relate to the configurable timestamp generated after applying any
```suggestion
The following variables relate to the configurable timestamp generated after applying any

```
```

* *Formatted*
+
The _Generate date and time.Formatted_ variable provides
the time stamp using the date and time format selected.
* *Date*
+
The _Generate date and time.Date_ variable provides the date.
* *Time* 
+
The _Generate date and time.Time_ variable provides the time.
* *Year* 
+
The _Generate date and time.Year_ variable provides the year.
* *Month* 
+
The _Generate date and time.Month_ variable provides the month
as a decimal number.
* *Day* 
+
The _Generate date and time.Day_ variable provides the day as a
decimal number.
* *Hour* 
+
The _Generate date and time.Hour_ variable provides the number
of hours.
* *Minute* 
+
The _Generate date and time.Minute_ variable provides the
number of minutes.
* *Second* 
+
The _Generate date and time.Second_ variable provides the
number of seconds.

== Wizard

The Wizard enables you to generate a timestamp based on the specified time variables, select the starting day of the week, and specify the output format and language. 

* *DateTime settings* 
+
Enables you to specify the date to generate. 
+
Select a *Generation mode* and specify the time variables to generate the desired date output. The available generation modes are: 

> ** *Current system date and time*: Uses the current system time as the timestamp.

Reviewer: timestamp is two different ways here. Best to make it consistent (either time stamp or timestamp--both are correct but I think timestamp is more commonly used in our world)

Reviewer: S/R throughout

** *Custom date and time*: Uses the time selected in the *DateTime* field as the timestamp.
** *Generated date and time from given variables*: Calculates the timestamp using the specified values. 

* *Workday settings* 
+
Enables you to specify a given day of the month by following some rules. 
+

> Select *Override date to* and use the dropdown options specify a day in the month to use.

Reviewer: ```suggestion
Select *Override date to* and use the dropdown options to specify a day in the month to use.

```
